openmc/element.py

```diff
@@ -102,9 +123,160 @@ def expand(self):
+                        a = int(nuclide[name_len:])
+                        i = min(list(range(len(nuclides_a))), key=lambda j: \
+                                abs(nuclides_a[j] - a))
+
+                        # Increment abundance of the nearest nuclide
+                        abundances[i] += abundance
+
+        # If a cross_section library is not present, expand the element into
+        # its natural nuclides
+        else:
+            for nuclide, abundance in sorted(NATURAL_ABUNDANCE.items()):
+                if re.match(r'{}\d+'.format(self.name), nuclide):
+                    nuclides.append(openmc.Nuclide(nuclide))
+                    abundances.append(abundance)
+
+        # Create a list of atomic masses
+        n_nuclides = len(nuclides)
+        atomic_masses = []
+        for nuclide in nuclides:
+            atomic_masses.append(openmc.data.atomic_mass(nuclide.name))
+
+        # Modify mole fractions if enrichment provided
+        if enrichment is not None:
+
+            # Get the indices for the uranium nuclides
+            for i,nuc in enumerate(nuclides):
+                if nuc.name == 'U234':
+                    u234 = i
+                elif nuc.name == 'U235':
+                    u235 = i
+                elif nuc.name == 'U238':
+                    u238 = i
+
+            # Calculate the mass fractions of isotopes
+            abundances[u234] = 0.008 * enrichment
+            abundances[u235] = enrichment
+            abundances[u238] = 1.0 - 1.008 * enrichment
+
+            # Convert the mass fractions to mole fractions
+            for i in range(n_nuclides):
+                abundances[i] /= atomic_masses[i]
+
+            # Normalize the mole fractions to one
+            sum_abundances = sum(abundances)
+            for i in range(n_nuclides):
+                abundances[i] /= sum_abundances
+
+        # Compute the ratio of the nuclide atomic massess to the element
+        # atomic mass
+        if percent_type == 'wo':
+
+            # Compute the element atomic mass
+            element_am = 0.
+            for i in range(n_nuclides):
+                element_am += atomic_masses[i] * abundances[i]
+
+            # Convert the molar fractions to mass fractions
+            for i in range(n_nuclides):
+                abundances[i] *= atomic_masses[i] / element_am
+
+            # Normalize the mass fractions to one
+            sum_abundances = sum(abundances)
+            for i in range(n_nuclides):
+                abundances[i] /= sum_abundances
+
+        # Create a list of the isotopes in this element
         isotopes = []
-        for isotope, abundance in sorted(NATURAL_ABUNDANCE.items()):
-            if re.match(r'{}\d+'.format(self.name), isotope):
-                nuc = openmc.Nuclide(isotope)
-                isotopes.append((nuc, abundance))
+        for nuclide, abundance in zip(nuclides, abundances):
+            pct = float('{:2.8g}'.format(percent*abundance))
```

Comment:
When I change it to `pct = percent * abundance` the `test_element_wo` and `test_enrichment` tests fail for Python 3. Any thoughts on how to avoid that? Perhaps cut off the amount when writing to the xml file?